lib/filewatch/discover.rb

```diff
@@ -0,0 +1,141 @@
+require_relative 'watched_file'
+
+module FileWatch
+  class Discover
+    attr_reader  :logger, :files, :wf_vars
```

Comment:
I could move these three vars (delimiter, close_older and ignore_older) into classes that encapsulates their function.